### PR TITLE
Rename serverlessMode to useCSIDriver

### PR DESCRIPTION
**Pull request by luhi-DT**

### CODE REVIEW

File: api/v1beta1/properties.go [hunk 1]
```diff
@@ -108,22 +108,22 @@ func (dk *DynaKube) ActiveGateImage() string {
 	return fmt.Sprintf("%s/linux/activegate:latest", registry)
 }
 
-func (dk *DynaKube) ServerlessMode() bool {
+func (dk *DynaKube) UseCSIDriver() bool {
```
Comment (chrismuellner) [inline api/v1beta1/properties.go:111]: Can we combine `UseCSIDriver()` and `NeedsCSI()` into a single function that is used to check if csi driver should be enabled?

Only the result of all the checks (`NeedsCSI()` currently) should be exposed and used in other parts of the code.

File: api/v1beta1/properties.go [hunk 2]
```diff
@@ -108,22 +108,22 @@ func (dk *DynaKube) ActiveGateImage() string {
...
-		return dk.Spec.OneAgent.ApplicationMonitoring.ServerlessMode
+		return dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver != nil && *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
 	}
 	return false
 }
 
 func (dk *DynaKube) NeedsCSI() bool {
-	if dk.ServerlessMode() {
-		return false
+	if dk.UseCSIDriver() {
```
Comment (chrismuellner) [inline api/v1beta1/properties.go:121]:
```suggestion
	return dk.UseCSIDriver() || dk.CloudNativeFullstackMode() || (dk.ApplicationMonitoringMode() && dk.Spec.OneAgent.ApplicationMonitoring.Image == "")
```
Does this have the same result?

File: api/v1beta1/properties.go [hunk 3]
```diff
@@ -108,22 +108,22 @@ func (dk *DynaKube) ActiveGateImage() string {
...
-		return dk.Spec.OneAgent.CloudNativeFullStack.ServerlessMode
+		return true
 	} else if dk.ApplicationMonitoringMode() {
-		return dk.Spec.OneAgent.ApplicationMonitoring.ServerlessMode
+		return dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver != nil && *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
 	}
 	return false
 }
 
 func (dk *DynaKube) NeedsCSI() bool {
```
Comment (chrismuellner) [inline api/v1beta1/properties.go:120]: Please add some unit tests for the possible combinations in `properties_test.go`

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: controllers/csi/provisioner/reconciler_test.go [hunk 4]
```diff
@@ -453,9 +451,7 @@ func TestHasCodeModulesWithCSIVolumeEnabled(t *testing.T) {
 			Spec: dynatracev1beta1.DynaKubeSpec{
```
Comment (chrismuellner) [inline controllers/csi/provisioner/reconciler_test.go:451]: please also rename the test from mentioning code modules/serverlessmode

File: api/v1beta1/properties.go (hunk 2 above)
Comment (luhi-DT) [reply api/v1beta1/properties.go:121]: would have the same result, but when merging UseCSIDriver and NeedsCSI, return statement will be pretty long

Comment (chrismuellner) [reply api/v1beta1/properties.go]: not really, it's mostly checking the same values

File: api/v1beta1/oneagent_types.go [hunk 5]
```diff
@@ -125,6 +125,9 @@ type ApplicationMonitoringSpec struct {
 	// Example: {major.minor.release} - 1.200.0
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OneAgent version",order=11,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:com.tectonic.ui:text"}
 	Version string `json:"version,omitempty"`
+
+	// Optional: If you want to use CSIDriver; disable if your cluster does not have 'nodes' to fall back to the volume approach.
+	UseCSIDriver *bool `json:"NeedsCSIDriver,omitempty"`
```
Comment (chrismuellner) [inline api/v1beta1/oneagent_types.go:130]:
```suggestion
	UseCSIDriver *bool `json:"needsCSIDriver,omitempty"`
```

**Review by chrismuellner — CHANGES_REQUESTED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**